src/ResourceManagement/Compute/VirtualMachineScaleSetMsiHelper.cs

```diff
@@ -0,0 +1,436 @@
+        private async Task<Microsoft.Azure.Management.Graph.RBAC.Fluent.IRoleAssignment> CreateRbacRoleAssignmentIfNotExistsAsync(IServicePrincipal servicePrincipal, string roleOrRoleDefinition, string scope, bool isRole, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string roleAssignmentName = SdkContext.RandomGuid();
+            try
+            {
+                if (isRole)
+                {
+                    return await rbacManager
+                        .RoleAssignments
+                        .Define(roleAssignmentName)
+                        .ForServicePrincipal(servicePrincipal)
+                        .WithBuiltInRole(BuiltInRole.Parse(roleOrRoleDefinition))
+                        .WithScope(scope)
+                        .CreateAsync(cancellationToken);
+                }
+                else
+                {
+                    return await rbacManager
+                        .RoleAssignments
+                        .Define(roleAssignmentName)
+                        .ForServicePrincipal(servicePrincipal)
+                        .WithRoleDefinition(roleOrRoleDefinition)
+                        .WithScope(scope)
+                        .CreateAsync(cancellationToken);
+                }
+            }
+            catch (CloudException cloudException)
+            {
+                if (cloudException.Body != null && cloudException.Body.Code != null && cloudException.Body.Code.Equals("RoleAssignmentExists", StringComparison.OrdinalIgnoreCase))
+                {
+                    // NOTE: We are unable to lookup the role assignment from principal.RoleAssignments() list
+                    // because role assignment object does not contain 'role' name (the roleDefinitionId refer
+                    // 'role' using id with GUID).
+                    return null;
+                }
+                throw cloudException;
+            }
+        }
+
+        /// <summary>
+        /// Given an object holding a numeric in Integer or String format, convert that to
+        /// Integer.
+        /// </summary>
+        /// <param name="obj">The object.</param>
+        /// <return>The integer value.</return>
+        ///GENMHASH:663112EB0F1DE517DFEE2A837DECC2B6:E55F34EE6CDFC6C04AA107D22EE30395
+        private static int? ObjectToInteger(object obj)
+        {
+            int? result = null;
+            if (obj != null)
+            {
+                if (obj is Int16)
+                {
+                    result = (int)((Int16)obj);
+                }
+                else if (obj is Int32)
+                {
+                    result = (int)obj;
+                }
+                else if (obj is Int64)
+                {
+                    result = (int)((Int64)obj);
+                }
+                else
+                {
+                    result = int.Parse((string)obj);
```

Comment:
>Parse [](start = 33, length = 5)

this will throw if the object is invalid int. there should be a method TryParse that should match your use case.